api/src/main/java/eu/darkbot/api/managers/ConfigAPI.java

```diff
@@ -14,7 +15,7 @@
 /**
  * Provides access to the bot user-managed configuration.
  */
-public interface ConfigAPI extends API.Singleton {
+public interface ConfigAPI extends API.Singleton, Dropdown.Options<String> {
```

Comment:
I'm not really sold on the idea that ConfigAPI should be itself the implementor of Dropdown.Options<String> for configs. Consider following the approach of MapNames and NpcNames (in the eu.darkbot.shared.config package), and just add a `ConfigNames` or similar.

This makes it so you can define a dropdown of config options like:

```java
@Dropdown(options = ConfigNames.class)
public String CONFIG_OPTION = "";
```
which makes more sense than "ConfigAPI"

Reply:
Config list is specific for each user so I not sure is it have sense to provide list to user code as is.

ConfigAPI is the root entry point for getting configs data and extracting dropdown list of existing configs from there is not weird because there happen config management and was added switching configs over itself list.

I think that creating not util or not interface classes in api repo is mistake and crutch because this implementation is not required for research while studding of creating plugins and I think that it should be hidden in main repo.

Because now it looks like moving `ConfigSupplier` implementation below to API repo with class name `ConfigName`.
![image](https://github.com/darkbot-reloaded/DarkBotAPI/assets/52008918/38db49c4-db94-4243-b43d-da5acfe3cf14)

My main idea was made clean interface way for getting list via small changing there.

Reply:
> Config list is specific for each user so I not sure is it have sense to provide list to user code as is.

Lots of info is specific for each user, the point of an API is to expose things that can be useful, you are not the only user of this API, and others may have different use-cases, where having access to this list may be useful.
One example: a weekly schedule table that wants to implement a dropdown to select from the list of available configs, it doesn't want a Dropdown.Options, it just wants a List of the available configs and it will have to implement it's own UI ontop of it to work inside a table.

> ConfigAPI is the root entry point for getting configs data and extracting dropdown list of existing configs from there is not weird because there happen config management and was added switching configs over itself list.
Config API is indeed the root entrypoint for interacting with the configuration, however the Dropdown.Options is a `UI` element, which should NOT be implemented by the logic component of configAPI. They're two diff things and should be kept separate, one is logic for how to access/manipulate configs, the other is how to render a list of configs to the ui.
And regarding that "ConfigSupplier" yes, the point is just to bring essentially that into api repo with a diff class name, because that's literally what's useful to do here, provide it as a shared utility that plugins can just use instead of having to implement their own

Reply:
Ok, I understood you not want mixing UI and API code. It have sense. 
If we creating new entity in API repo than can we create interfaces in `eu.darkbot.api.config.types` package
```java
interface ConfigProfile { 
  boolean apply();
}

interface BotModule {
  boolean apply();
}
```
and hardcoding editors for each interface?